experimental/canvas/graphql/readme.md

```diff
@@ -0,0 +1,164 @@
+are nested pages (e.g. how do you page through enrollments when you also
+have paging turned on for courses?)
+
+The notes below show _most_, but perhaps not _all_ of the individual elements
+that need to be requested. Further detailed analysis will need to be carried out
+while looking at the mapping of Canvas API elements to the CSV file extracts.
+
+* Assignments - YES
+
+  ```none
+  query MyQuery {
+    account(id: "1") {
+      coursesConnection {
+        nodes {
+          _id
+          name
+          courseCode
+          assignmentsConnection {
+            nodes {
+              _id
+              dueAt
+              dueDateRequired
+              name
+              pointsPossible
+              submissionTypes
+            }
+          }
+        }
+      }
+    }
+  }
+  ```
+
+* Authentication Events (treated as "System Activities") - NO, unless overlooked
+* Grades - YES
+
+  ```none
+  query MyQuery {
+    account(id: "1") {
+      coursesConnection {
+        nodes {
+          enrollmentsConnection {
+            nodes {
+              user {
+                enrollments {
+                  grades {
+                    finalScore
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+  ```
+
+* Enrollments (treated as "Section Associations") - YES, see `graphql.py`.
+* Sections - YES
+
+  ```none
+  query MyQuery {
+    account(id: "1") {
+      coursesConnection {
+        nodes {
+          _id
+          name
+          sectionsConnection {
+            nodes {
+              _id
+              name
+              sisId
+            }
+          }
+          courseCode
+        }
+      }
+    }
+  }
+  ```
+
+* Submissions - YES
+
+  ```none
+  query MyQuery {
+    account(id: "1") {
+      coursesConnection {
+        nodes {
+          _id
+          name
+          courseCode
+          submissionsConnection {
+            nodes {
+              _id
+              createdAt
+              grade
+              gradingStatus
+              late
+              missing
+              postedAt
+              score
+              state
+              user {
+                _id
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+  ```
+
+* Users - YES, see `graphql.py`.
+
+## Next Steps
+
+1. More experimentation with paging for _nested_ resources. Paging should be set at 100.
+2. Combine into one big query.
+3. Write that big GraphQL query and the paging process into the Canvas Extractor,
+   perhaps in the API folder or directly in `client_facade`.
+4. Carefully review all existing logic and bring it into the query, for example:
+   * Which columns to retrieve and map into the UDM
+   * Any filtering on teachers, status
+   * Course dates
+5. Use opnieuw on GraphQL query for rate limit handling.
+6. Remove old API folder and `client_facade` code, except for System Activities.
```

Comment:
For LMS-388 - the next performance spike ticket - I will do a little more research on these notes and turn them into one or more tickets.